### PR TITLE
Feature update u icompiler

### PR DESCRIPTION
**Pull request by JRojas999**

### CODE REVIEW

File: src/main/java/org/fundacion/jala/converter/view/MainInterface.java [hunk 1]
```diff
@@ -204,4 +205,8 @@ public void mouseExited(final MouseEvent eventListener) {
         button.addMouseListener(mouseListener);
         return button;
     }
+
+    public static void main(String[] args) {
+        new MainInterface().initInterface("hoa");
```
Comment (JessickaMoya) [inline src/main/java/org/fundacion/jala/converter/view/MainInterface.java]: What is "hoa"?

File: src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java [hunk 2]
```diff
@@ -133,4 +138,39 @@ public void JTreePanelOne(final String vals[]) {
...
+        treeModel = new DefaultTreeModel(root);
+        treeOne = new JTree(treeModel);
+        for (String fileName : fileNames) {
+            root.add(new DefaultMutableTreeNode(fileName));
+        }
+        treeOne.setShowsRootHandles(true);
+    }
+
+    /**
+     * Adds a child path that represents new file of project.
```
Comment (JessickaMoya) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java:162]: javadoc missing @param

File: src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java [hunk 3]
```diff
@@ -133,4 +138,39 @@ public void JTreePanelOne(final String vals[]) {
...
+     */
+    public void createChild(final String nameFile) {
+        root.add(new DefaultMutableTreeNode(nameFile));
+        treeModel.reload();
+    }
+
+    /**
+     * Removes a child path that represents new file of project.
```
Comment (JessickaMoya) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java:172]: javadoc missing @param

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java [hunk 4]
```diff
@@ -88,19 +87,14 @@ public void mouseClicked(final MouseEvent mouseEvent) {
...
-                tabList.add(title);
-                CodeTextArea codeArea = new CodeTextArea();
-                codeArea.setName(title);
-                codeArea.getCodeArea().setText(InitialCode.generate(title, extension));
-                add(codeArea, getTabCount() - 1);
-                setTabComponentAt(getTabCount() - 2, createTabHeader(title));
-                setSelectedIndex(getTabCount() - 2);
+                addProjectTab(title, InitialCode.generate(title, extension));
+                updateCodeTextAreas(title);
+//                codeTextAreas.get(0).createChild(title);
```
Comment (JessickaMoya) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java]: Delete commented line

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java [hunk 5]
```diff
@@ -224,6 +219,64 @@ public void setExtension(final String fileType) {
...
+    public void updateCodeTextAreas(final String fileName) {
+        for (int i = 0; i < codeTextAreas.size() - 1; i++) {
+            codeTextAreas.get(i).createChild(fileName);
+        }
+    }
+
+    /**
+     * Deletes a file form all tree tabs.
+     */
+    public void deleteCodeTextAreas(final int index) {
```
Comment (Paola-Xaq) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:282]: @parameter is missing in JavaDoc

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java [hunk 6]
```diff
@@ -224,6 +219,64 @@ public void setExtension(final String fileType) {
...
+        CodeTextArea codeArea = new CodeTextArea(projectName, tabList);
+        codeArea.setName(title);
+        codeArea.getCodeArea().setText(initialCode);
+        codeTextAreas.add(codeArea);
+    }
+
+    /**
+     * Updates file tree of all tabs.
+     */
+    public void updateCodeTextAreas(final String fileName) {
```
Comment (Paola-Xaq) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:271]: @ parameter is missing in JavaDoc

File: src/main/java/org/fundacion/jala/converter/view/MainInterface.java (hunk 1 above)
Comment (JRojas999) [reply src/main/java/org/fundacion/jala/converter/view/MainInterface.java]: Sorry, I forgot to delete that main.

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java (hunk 4 above)
Comment (Paola-Xaq) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java]: Commented lines should not be push

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java [hunk 7]
```diff
@@ -33,34 +32,35 @@
...
-        add(emptyInitial);
+        tabList = new ArrayList<>();
+        codeTextAreas = new ArrayList<>();
+        projectName = "projectexample";
+        addInitialProjectTab("Initial", "Create a Project");
     }
 
     /**
      * Starts required components to add new tabs.
      */
```
Comment (Paola-Xaq) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:58]: @ param are missing in JavaDoc

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java [hunk 8]
```diff
@@ -218,20 +210,20 @@ public int changeLanguage() {
...
+            try {
+                LOGGER.info("Execute Try");
+                String endpointResponse = clientRequest.executeRequest(projectRequestForm, token);
+                projectId = endpointResponse;
+                consoleOutput.getConsole().setText("Project " + projectTitle + " was created successfully");
+                projectTab.start(extension, projectTitle);
+                LOGGER.info("finish");
+            } catch (Exception exception) {
+                LOGGER.error("Execute Exception" + exception.getMessage());
+            }
```
Comment (Paola-Xaq) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java]: I Think the tab space in this lines, should be removed? doesn't look like the correct tabulation

Comment (JRojas999) [reply src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java]: You're right, my mistake.

File: src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java (hunk 2 above)
Comment (JRojas999) [reply src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java:162]: I will complete it.

File: src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java (hunk 3 above)
Comment (JRojas999) [reply src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java:172]: I will complete it.

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java (hunk 4 above)
Comment (JRojas999) [reply src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java]: I forgot it. 😅

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java (hunk 7 above)
Comment (JRojas999) [reply src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:58]: I will complete it.

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java (hunk 6 above)
Comment (JRojas999) [reply src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:271]: I will complete it.

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java (hunk 5 above)
Comment (JRojas999) [reply src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:282]: I will complete it.

**Review by JessickaMoya — APPROVED** [review]

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java [hunk 9]
```diff
@@ -218,20 +210,20 @@ public int changeLanguage() {
     public void createProject() {
         LOGGER.info("start");
         ProjectRequestForm projectRequestForm = new ProjectRequestForm();
-        String result = (String) JOptionPane.showInputDialog(null,
+        String projectTitle = (String) JOptionPane.showInputDialog(null,
                 "Write project name", "Project Name",
                 JOptionPane.PLAIN_MESSAGE, null, null, "project1");
-        projectRequestForm.addProjectName(result);
+        projectRequestForm.addProjectName(projectTitle);
         projectRequestForm.addUserId("1");
```
Comment (Paola-Xaq) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java:217]: Don't know if it is considered a magic number the 1 in the addUserId?

**Review by Paola-Xaq — APPROVED** [review]
Just one comment missing, that I don't know if it is necessary, else, everything seems ok. Good Job.

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java [hunk 10]
```diff
@@ -139,7 +137,6 @@ public CodeTextArea getSelectedPane() {
      * @return a JPanel with complete title.
      */
     public JPanel createTabHeader(final String title) {
-
         ImageIcon addIcon = new ImageIcon("img/compilerBtn/BtnCloseTab.png");
```
Comment (SaulCM) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:140]: Image's path could be in a final attribute.

**Review by SaulCM — APPROVED** [review]

**Pull request merged**